lib/ansible/modules/cloud/ovirt/ovirt_disk.py

```diff
@@ -690,6 +713,18 @@ def main():
             elif state == 'detached':
                 ret = disk_attachments_module.remove()
 
+        # When the host parameter is specified and the disk is not being
+        # removed, refresh the information about the LUN.
+        if state != 'absent' and host:
+            hosts_service = connection.system_service().hosts_service()
+            hosts = hosts_service.list(search='name={0!s}'.format(host))
+            if len(hosts) > 0:
+                host_service = hosts[0]
+            else:
+                module.fail_json(msg='Host "{0!s}" does not exist'.format(host))
+
+            disks_service.disk_service(disk.id).refresh_lun(host_service)
```

Comment:
Would be better if you write it as( it will auto fail if host is not found):

```python
hosts_service = connection.system_service().hosts_service()
host_id = get_id_by_name(hosts_service, host)
disks_service.disk_service(disk.id).refresh_lun(otypes.Host(id=host_id))
```

Don't forget to add `from ansible.module_utils.ovirt import get_id_by_name`

Reply:
Thanks. This looks much nicer.